package/libpiksi/libpiksi/include/libpiksi/kmin.h

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+/**
+ * @file    kmin.h
+ *
+ * @brief   Find the K minimum (or maximum) elements of a group
+ *
+ * @defgroup    kmin
+ * @addtogroup  kmin
+ * @{
+ */
+
+#ifndef LIBPIKSI_KMIN_H
+#define LIBPIKSI_KMIN_H
+
+#include <libpiksi/common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief The sort algorithm to use for finding K-minimum values
+ */
+typedef enum {
+  KMIN_ALGORITHM_STDLIB_QSORT,   /**< C stdlib qsort */
+  KMIN_ALGORITHM_KLIB_INTROSORT, /**< klib introsort */
+  KMIN_ALGORITHM_MACRO_TIMSORT,  /**< The "timsort" (from Python) algorithm, fastest for partially
+                                    ordered data. */
+  KMIN_ALGORITHM_MACRO_QSORT, /**< A macro version of quick sort, slightly faster than the built-in
```

Comment:
The idea was to enable a number of algorithms so we could pick whatever is the fastest for the the particular dataset that's being used as input... this is probably a premature optimization though, since I don't have any evidence (yet) that we have data sets that would benefit from one sorting algorithm vs another (nor evidence that it matters from the perspective of the problem that this PR is trying to solve).  So, realistically we can probably just dump this algorithm selection for now and use one sorting algorithm (timsort) since it seems to be the fastest for the datasets that `kmin_t` is currently used on.

Reply:
Removed these, but left it as a commit so we can revive the code if needed.